noxfile.py

```diff
@@ -1,55 +1,126 @@
+# /// script
+# requires-python = ">=3.8"
+# dependencies = [
+#     "nox==2025.5.01",
+# ]
+# ///
+
 # SPDX-License-Identifier: MIT
 
 from __future__ import annotations
 
 import os
 import pathlib
-from itertools import chain
-from typing import TYPE_CHECKING, Callable, Dict, List, Tuple, TypeVar
+import sys
+from typing import Any, Dict, List, Sequence, Tuple
 
 import nox
 
-if TYPE_CHECKING:
-    from typing_extensions import Concatenate, ParamSpec
-
-    P = ParamSpec("P")
-    T = TypeVar("T")
+nox.needs_version = ">=2025.5.1"
 
-    NoxSessionFunc = Callable[Concatenate[nox.Session, P], T]
+PYPROJECT = nox.project.load_toml()
 
 
-# see https://pdm-project.org/latest/usage/advanced/#use-nox-as-the-runner
-os.environ.update(
-    {
-        "PDM_IGNORE_SAVED_PYTHON": "1",
-    },
-)
+def use_min_python_of(version: str) -> str | None:
+    """Use the minimum necessary python for this run, but if the environment is a specific venv, then use that one."""
+    major, minor = version.split(".")
+    if sys.version_info < (int(major), int(minor)):
+        # If the current Python version is less than the required version, use the required version
+        return f"{major}.{minor}"
+    return None
 
 
 nox.options.error_on_external_run = True
-nox.options.reuse_existing_virtualenvs = True
+nox.options.reuse_venv = "yes"
+nox.options.default_venv_backend = "uv|virtualenv"
 nox.options.sessions = [
     "lint",
     "check-manifest",
     "slotscheck",
     "pyright",
     "test",
 ]
-nox.needs_version = ">=2022.1.7"
 
 
 # used to reset cached coverage data once for the first test run only
 reset_coverage = True
 
+EMPTY_SEQUENCE: Sequence[str] = set()  # type: ignore
```

Comment:
Why use a set here, and not an empty tuple?
```suggestion
EMPTY_SEQUENCE: Sequence[str] = ()
```